### PR TITLE
596 postgres feature to work

### PR DESCRIPTION
**Pull request by andreievg**
Closes: #596 

Sqlite default features was still enabled when using `--features postgres` flag. So though to do a check on postgres flag rather than sqlite flag.

### CODE REVIEW

**Review by clemens-msupply — APPROVED** [review]
Thanks!

File: repository/src/database_settings.rs [hunk 1]
```diff
@@ -29,7 +29,7 @@ impl DatabaseSettings {
     }
 }
 
-#[cfg(feature = "sqlite")]
+#[cfg(not(feature = "postgres"))]
```
Comment (mark-prins) [inline repository/src/database_settings.rs:32]: commenting as a non-rust developer! so I don't know the implications of what I'm saying and you can happily ignore.

having this logic in so many places in the code feels brittle - and requires a lot of file changes. Is it not possible to have this check once, and set a global flag for `database_engine` or similar?

Would also help to isolate the implicit configuration here, of `not(postgres)` = `sqlite`

Comment (clemens-msupply) [reply repository/src/database_settings.rs:32]: Yes it's brittle, and everything to save a dynamic dispatch... Not aware of an easier solution, probably you can write a macro for it ;-)

**Comment by mark-prins** [comment]
tested and working on windows!

**Pull request merged**